### ISSUE TITLE
Add support for HTTP basic authentication

### ISSUE THREAD

**Issue by GentlemanHal** [build-canaries/nevergreen#38]

**Comment by GentlemanHal** [comment]
Depends on issue #33

**Comment by joejag** [comment]
java.util.Base64 is only in Java8: http://docs.oracle.com/javase/8/docs/technotes/guides/lang/enhancements.html

Heroku and SnapCI happen to use Java8, but most people don't. Can we use a library instead?

**Comment by joejag** [comment]
If you try to use the auth on the Apache website it returns a 401 but we return a 500 to the user. I suspect the status codes are important for this feature.

I'll open a new ticket to add this feature.

**Comment by joejag** [comment]
I would of thought we'd encrypt the username + password. Not massive problem. Just not what I expected

**Comment by joejag** [comment]
Bug: You can corrupt the password by hitting fetch twice with the password entry dialog open.

Quick Fix: Close the password dialog when the fetch button is pressed.
Better Fix?: check to see if the password field has been changed (after first set, set the password field to known dummy value and see if it isn't that on saving the password)

**Comment by joejag** [comment]
Not mandatory:

In general when sending username+password it is best sending it in the headers. URLs get logged a lot. Like we do here: 
``` 
INFO: Request start: :get /api/projects url=http%3A%2F%2Flocalhost%3A5000%2Ftest_data.xml&serverType=unknown&username=a&password=MVypDKr1MKF2v7z5Kc7%2BXp%2BYkoPLUsGUg%2B%2BAXf38qg8%3D
```

Of the URL, body and headers. Headers tend to be logged the least.

I don't think we need to change Nevergreen for this.

**Comment by joejag** [comment]
The actual username + password save, encrypt, base64 stuff all seems to work. Good stuff.

**Comment by cowley05** [comment]
Thanks joe, thought i had fixed that but looks like i un fixed it with a later push haha

will sort out the double fetch. I will also take a look at refactoring that bit of code to put it in the header but no promises ;)

**Comment by joejag** [comment]
Cool. No worries about the header thing.

**Comment by cowley05** [comment]
Joe i fixed that issue, if you have time do you want to skim over it? if not i can close...

**Comment by joejag** [comment]
QA: :+1:
